rocks-lib/src/build/mod.rs

```diff
@@ -32,7 +33,35 @@ fn install(
     Ok(())
 }
 
-pub fn build(rockspec: Rockspec, config: &Config) -> Result<()> {
+#[async_recursion]
+pub async fn build(rockspec: Rockspec, config: &Config) -> Result<()> {
+    // TODO(vhyrro): Create a unified way of accessing the Lua version with centralized error
+    // handling.
+    let lua_version = rockspec.lua_version();
+    let lua_version = config
+        .lua_version()
+        .or(lua_version.as_ref())
+        .ok_or_eyre(
+            "lua version not set! Please provide a version through `--lua-version <ver>`",
+        )?;
+
+    let tree = Tree::new(config.tree().clone(), lua_version.clone())?;
+
+    // Recursively build all dependencies.
+    // TODO: Handle build dependencies as well.
+    for dependency_req in rockspec
+        .build_dependencies
+        .current_platform()
+        .iter()
+        .filter(|req| tree.has_rock(req).is_none())
+    {
+        // NOTE: This recursive operation will create another `tree` object
+        // which will in turn acquire another lock to the filesystem.
+        // Once we implement fs locks, this could become a problem, so a more sophisticated
+        // filesystem acquiring mechanism will have to be devised.
```

Comment:
question: a more sophisticated fs acquiring mechanism than what?
Afaik, we don't acquire anything yet?

Reply:
Forgive my 11pm brain for that one. While we don't acquire any "locks", there will be many tree objects and that could very quickly become a problem with concurrent writes. In this case it's fine, because we always await each installation operation to finish before starting the next one.